unittests/data/type_traits.hpp

```diff
@@ -544,13 +586,14 @@ namespace yes{
     struct x{
         x(){}
     };
+#ifdef __castxml__
+    typedef details::const_item const_item;
+    typedef details::const_container const_container;
+#endif
```

Comment:
These are being detected as having trivial constructors by CastXML, but I don't think that's correct. `const_item` and `const_container` are both not default constructible because they have `const` array members. 

The weird thing is `std::is_trivially_constructible` returns `false` like I expect, but `boost::has_trivial_constructor` returns `true`, [example](http://coliru.stacked-crooked.com/a/2dd0650edca00c82). I'll look into this too later.